examples/lightning_base.py

```diff
@@ -39,6 +40,18 @@
 }
 
 
+# update this to support new schedulers from transformers.optimization
+# the listed below schedulers appear in the output of --lr_scheduler=help
+arg_to_scheduler = {
+    "linear": "get_linear_schedule_with_warmup",
+    "cosine": "get_cosine_schedule_with_warmup",
+    "cosine_w_restarts": "get_cosine_with_hard_restarts_schedule_with_warmup",
```

Comment:
consider importing the functions you use and then hardcoding
```python
arg_to_scheduler = {"linear": get_linear_schedule_with_warmup, ...}
```
That way IDEs can find usages.

Reply:
+1 on this

Reply:
I used the single source approach, to avoid maintaining multiple entries of the same. So that the import happens on demand.

With your suggestion, we need to repeat each entry twice - import + populate dict (and to do =help, need to derive the function name from the object, but oh well, that's easy).

An alternative solution is to import `transformers.optimization` and then use the full form in the dic:
```
import transformers.optimization
arg_to_scheduler = {
    "linear": transformers.optimization.get_linear_schedule_with_warmup,
    "cosine":  transformers.optimization.get_cosine_schedule_with_warmup,
    ...
```